guacamole/src/main/java/org/apache/guacamole/rest/auth/AuthenticationService.java

```diff
@@ -93,12 +93,12 @@
     /**
      * Regular expression which matches any IPv4 address.
      */
-    private static final String IPV4_ADDRESS_REGEX = "([0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3})";
+    private static final String IPV4_ADDRESS_REGEX = "([0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3})(?::[0-9]{1,5})?";
```

Comment:
The documentation for this constant currently states:

> Regular expression which matches any IPv4 address.

Modifying this constant such that it also matches port numbers isn't complete in itself. That change would need to be followed through such that the documentation is correct. The same goes for the naming of the constant (`IPV4_ADDRESS_REGEX` is pretty specific to IPv4 addresses).

It may be much simpler to modify `IP_ADDRESS_REGEX` instead. This would remove the duplicated pattern and may be easier to rename and document.

Reply:
Another way may would be to create e new constant PORT_NUMBER_REGEX. And combine it within the X_FORWARDED_FOR Constant. Then you have an impended Port Regex and the renaming can be avoided.

Reply:
Sure. That sounds good and satisfies the criteria.